core/modules/field/field.block.inc

```diff
@@ -80,9 +80,34 @@ class FieldBlock extends Block {
     list($entity_type, $field_name) = explode('-', $this->childDelta);
     $entity = $this->contexts[$entity_type]->data;
 
-    // Do not render if the entity is not found on this layout.
-    if (empty($entity)) {
-      return NULL;
+    // Do not proceed if the entity is not an EntityInterface object
+    if (!is_a($entity, 'EntityInterface')) {
+      if ($entity_type == 'user') {
+        if (isset($entity->uid)) {
+          // Get a complete user entity.
+          $context_info = layout_get_context_info('user');
+          // Check if a load function has been set by some other module. If not, default to user_load().
+          if (isset($context_info['load callback']) && function_exists($context_info['load callback'])) {
```

Comment:
Not exactly sure why this loop is necessary TBH. `load callback` is required in `hook_layout_context_info()` isn't it?

Reply:
oh, so it is. not only that, but the load hook function and the key name are both available from entity_get_info(), so this can stop being specific to user entities.